dune_client/client_async.py

```diff
@@ -0,0 +1,152 @@
+""""
+Async Dune Client Class responsible for refreshing Dune Queries
+Framework built on Dune's API Documentation
+https://duneanalytics.notion.site/API-Documentation-1b93d16e0fa941398e15047f643e003a
+"""
+import asyncio
+from typing import Any
+
+from aiohttp import ClientSession, ClientResponse, ContentTypeError, TCPConnector, ClientTimeout
+
+from dune_client.base_client import BaseDuneClient
+from dune_client.models import (
+    ExecutionResponse,
+    DuneError,
+    ExecutionStatusResponse,
+    ResultsResponse,
+    ExecutionState,
+)
+
+from dune_client.query import Query
+
+
+# pylint: disable=duplicate-code
+class AsyncDuneClient(BaseDuneClient):
+    """
+    An asynchronous interface for Dune API with a few convenience methods
+    combining the use of endpoints (e.g. refresh)
+    """
+    _session = None
```

Comment:
Putting the session here as None turns the type into an `Optional[ClientSession]` and causes a lot of hassle when trying to access session throughout the code. Is there any particular reason it needs to be defined out here initially as None? I believe all these issues would go away if it were merely defined in the constructor.

Reply:
Yeah, that makes sense. Let me think if we can remove None here